src/effects/builtin/noisegen/noiseviewmodel.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Audacity: A Digital Audio Editor
+ */
+#include "noiseviewmodel.h"
+
+#include "log.h"
+
+using namespace au::effects;
+
+QStringList NoiseViewModel::typeNames() const
+{
+    static const QStringList names {
+        tr("White", "not a color, but 'white noise' having a uniform spectrum"),
+        tr("Pink", "not a color, but 'pink noise' having a spectrum with more power in low frequencies"),
+        tr("Brownian", "a kind of noise spectrum also known as 'red' or 'brown'"),
+    };
+
+    return names;
+}
+
+QVariantMap NoiseViewModel::types() const
```

Comment:
I recommend returning a list of items here, something like that
```
QVariantList NoiseViewModel::types() const
{
       QVariantList items; 
       if (items.empty()) {
               QVariantMap white; 
               white["text"] = tr("White", "not a color, but 'white noise' having a uniform spectrum");
               white["value"] = (int)NoiseEffect::Type::White;
               items << white;

               ...
       }
       return items;
}      
```

And then this list of items can be specified in the combo box as a model
```
ComboBox {
            id: typeSelector
            model: noise.types()
            textRole: "text"
            valueRole: "value"
}
```

Reply:
Thanks! That what I was looking for!